flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/scala/StreamTableEnvironment.scala

```diff
@@ -231,4 +231,20 @@ class StreamTableEnvironment(
   : Unit = {
     registerAggregateFunctionInternal[T, ACC](name, f)
   }
+
+  /**
+    * Registers a [[TableAggregateFunction]] under a unique name in the TableEnvironment's catalog.
+    * Registered [[TableAggregateFunction]] functions can only be referenced in Table API.
+    *
+    * @param name The name under which the function is registered.
+    * @param f The TableAggregateFunction to register.
+    * @tparam T The type of the output value.
+    * @tparam ACC The type of aggregate accumulator.
+    */
+  def registerFunction[T: TypeInformation, ACC: TypeInformation](
```

Comment:
This method can be removed and reuse registerFunction(name: String, f: AggregateFunction[T, ACC])) by changing the signature to registerFunction(name: String, f: UserDefinedAggregateFunction[T, ACC]))

Reply:
Yes, it would be good if we reuse it. However, It may not good to expose `UserDefinedAggregateFunction` to users. Besides, changing to `UserDefinedAggregateFunction` also makes it capable to register a class that extends directly from `UserDefinedAggregateFunction` which should be avoided.

Reply:
Make sense.